llama_stack/apis/inference/inference.py

```diff
@@ -1007,6 +1029,26 @@ async def embeddings(
         """
         ...
 
+    @webmethod(route="/inference/rerank", method="POST", experimental=True)
+    async def rerank(
+        self,
+        model: str,
+        query: str,
+        documents: list[str],
+        top_n: int | None = None,
```

Comment:
FWIW in the rest of the vector io code base we use top_k, so it may be intuitive to follow that pattern. Though they're not exactly one to one, so I could see an argument against it.

https://github.com/meta-llama/llama-stack/blob/main/llama_stack/providers/inline/vector_io/sqlite_vec/sqlite_vec.py

Reply:
top_n is,
- an optimization on output generation
- a common first operation a caller will perform

does None / not provided mean there will be an index,score pair for each input document?